### ISSUE TITLE
[arch needed] Change the way tabbing is tested for A11Y

### ISSUE THREAD

**Issue by Zeyber** [SAP/spartacus#13686]
Currently we test the ability to tab through a page by writing test configurations stemming from `tabbing-order.e2e-spec.ts`. This method has some issues:
- Every feature requires writing new tests.
- Every change to the DOM requires modifying the test configuration.
- Writing test configurations is tedious.
- Maintaining tests seems difficult to maintain. It frequently ends up with the flaky status and gets skipped in our release process or pipeline.

We could change this method for testing tabbing to address these issues.

Proposed Solution:
- We can create a new cypress method such as `cy.tabScreenshot()` which can take a screenshot at any point it is called in an e2e test.
- It can verify the tabbing journey through a page and fail the test if there are changes in the DOM (accessible elements via tab key) so that the developer can verify tabbing works with the changes.
- The verification will refer to a file storing the tab-able elements from a previous tab screenshot.
- `cy.tabScreenshot(scenario: string)` refers to the given scenario (eg. `cy.tabScreenshot('home-page')`).
-  The method can also generate image screenshot/s to show each tab-able element on the DOM at a given point for the dev to verify which elements are accessible via tab key at the given point.
- When verified, dev can push new `tab-states/home-page.json` generated by `cy.tabScreenshot('home-page') to repo with new state of elements at given point.

Advantages:
- Dev just needs to call `cy.tabScreenshot()` to test tabbing on a page and push generated .json files.
- No need to write and update more complex test configs every DOM change.
- Image screenshots for checking visual states of each tab key press.
